backend/api/submissions/types.py

```diff
@@ -11,15 +13,27 @@
 from .permissions import CanSeeSubmissionDetail, CanSeeSubmissionPrivateFields
 
 
-def restricted_field():
+def restricted_field() -> StrawberryField:
     """Field that can only be seen by admin, the submitter or who has the ticket
     until voting is not closed, after it will be public"""
-    return strawberry.field(permission_classes=[CanSeeSubmissionDetail])
 
+    def resolver(self, info: Info):
+        if CanSeeSubmissionDetail().has_permission(self, info):
+            return getattr(self, info.python_name)
+        return None
 
-def private_field():
+    return strawberry.field(resolver=resolver)
+
+
+def private_field() -> StrawberryField:
     """Field that can only be seen by admin and the submitter"""
-    return strawberry.field(permission_classes=[CanSeeSubmissionPrivateFields])
+
+    def resolver(self, info: Info):
+        if CanSeeSubmissionPrivateFields().has_permission(self, info):
```

Comment:
Can you fix https://github.com/pythonitalia/pycon/blob/5e3d8be0418ac70213112e000c306a3100ef3434/backend/api/submissions/permissions.py#L45 in this PR too?

So private_field are supposed to be seen only by admin and submitter, but right now there is:

```python
        if source.schedule_items.exists():  # pragma: no cover
            return True
```

which means that once it is added to the schedule you can fetch the private notes.

You only need to remove that `if`, if you can do it in this PR it would be great :)

Reply:
actually we might need to move that check into `CanSeeSubmissionDetail`

Reply:
@estyxx can you do this too? ^